Source/Bogus/Extensions/Australia/ExtensionsForAustraliaRegistrationPlate.cs

```diff
@@ -0,0 +1,175 @@
+         new PlateFormatRule { StartYear = 1940, EndYear = 1953, Format = "LL-NNN" },
+         new PlateFormatRule { StartYear = 1954, EndYear = 2013, Format = "LLL-NNN" },
+         new PlateFormatRule { StartYear = 2014, EndYear = 9999, Format = "NLL-NLL" }
+      ],
+      ["WA"] =
+      [
+         new PlateFormatRule { StartYear = 1956, EndYear = 1978, Format = "LLL-NNN" },
+         new PlateFormatRule { StartYear = 1979, EndYear = 9999, Format = "NLLL-NNN" }
+      ]
+   };
+
+
+   /// <summary>
+   /// This method generates a random registration date for a vehicle within a specified date range.
+   /// </summary>
+   /// <param name="vehicle"></param>
+   /// <param name="from"></param>
+   /// <param name="to"></param>
+   /// <returns></returns>
+   private static DateTime GenerateRegistrationDate(Vehicle vehicle, DateTime from, DateTime to)
+   {
+      // Swap if needed
+      if (from > to)
+      {
+         var temp = from;
+         from = to;
+         to = temp;
+      }
+      from = from.Date;
+      to = to.Date;
+      int duration = (int)(to - from).TotalDays;
+      int offset = vehicle.Random.Int(0, duration);
+      return from.AddDays(offset);
+   }
+
+   private static string GenerateRegistrationPlate(Vehicle vehicle, DateTime registrationDate, string state)
+   {
+      state = state.ToUpperInvariant();
+      if (!StatePlateRules.ContainsKey(state))
+         throw new ArgumentException($"Unsupported Australian state: {state}", nameof(state));
+      int year = registrationDate.Year;
+      var rules = StatePlateRules[state];
+      var rule = rules.FirstOrDefault(r => year >= r.StartYear && year <= r.EndYear);
+      if (rule == null)
+         throw new ArgumentException($"No plate format for {state} in year {year}");
+      return rule.Generator != null ? rule.Generator(vehicle, registrationDate) : GeneratePlateFromFormat(vehicle, rule.Format);
+   }
+
+   private static readonly char[] Letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".ToCharArray();
+   private static readonly char[] Digits = "0123456789".ToCharArray();
+
+   private static string GeneratePlateFromFormat(Vehicle vehicle, string format)
+   {
+      var sb = new StringBuilder();
+      foreach (var c in format)
+      {
+         switch (c)
+         {
+            case 'L': sb.Append(vehicle.Random.ArrayElement(Letters)); break;
+            case 'N': sb.Append(vehicle.Random.ArrayElement(Digits)); break;
+            case '0': sb.Append(vehicle.Random.Number(1)); break;
+            case 'M': sb.Append('M'); break;
+            case 'S': sb.Append('S'); break;
+            case 'Y': sb.Append('Y'); break;
+            case '-': sb.Append('-'); break;
+            default: sb.Append(c); break;
+         }
+      }
+      return sb.ToString();
+   }
+
+   private static string GenerateRandomState()
+   {
+      var randomizer = new Randomizer();
+      return randomizer.ListItem(SupportedStates.ToList());
```

Comment:
I don't think we should be creating a `new Randomizer()` here. Try pulling the `vehicle.Random` instead to pull a randomizer that has already been instantiated. We need generally need to re-use the randomizer that has already been created at start up to preserve determinsitc results.

Reply:
See this for example:
* https://github.com/bchavez/Bogus/blob/2e84969c556058437bd4ac9dce2edc7209339e7f/Source/Bogus/Extensions/UnitedKingdom/ExtensionsForGreatBritainRegistrationPlate.cs#L150